src/game/services/Lookup/index.ts

```diff
@@ -0,0 +1,31 @@
+import * as cards from '../../cards'
+import { isCardId } from '../../types/guards'
+import { ICard, IGame, IPlayer } from '../../types'
+import { GameStateCorruptError } from '../Rules/errors'
+
+export class Lookup {
+  static getCardFromHand(
```

Comment:
what is the benefit from a class with static methods here vs individual functions similar to utils?

Reply:
The idea is to avoid accumulating a collection of unrelated "util" functions. We've seen this problem manifest to a significant degree in OG Farmhand: https://github.com/jeremyckahn/farmhand/blob/develop/src/utils/index.js

We've made great progress with breaking those functions out into individual files, but even when that effort is complete we will be left a large, flat list of modules. At scale, that becomes hard to manage because it's a cluttered monolith of unrelated functions. This pattern achieves a sort of functional namespacing. At this small scale, it is admittedly overhead with minimal benefit. However, as we need more "lookup" type functionality we'll have a logical place to put the implementation. More importantly, we'll know where to find it later because the functionality is semantically colocated.

This is something I learned from a former coworker who's got very strong architectural skills. It seemed excessive to me at first, but I've come to really appreciate the pattern as a code base grows. 🙂